liota/dccs/iotcc.py

```diff
@@ -513,8 +540,16 @@ def store_reg_entity_attributes(self, entity_type, entity_name, reg_entity_id,
                 new_prop_dict = tmp_dict
         else:
             tmp_dict = self.read_entity_file(reg_entity_id)
-            if ((tmp_dict["entity type"] == entity_type) and (tmp_dict["name"] == entity_name)
-                and (tmp_dict["device type"] == dev_type)):
+            # check Entity_Timestamp of entity_file: if < boottime, get properties from cloud
+            if (self.enable_reboot_getprop == "True") and ('Entity_Timestamp' in tmp_dict):
+                last_dtime = datetime.datetime.strptime(tmp_dict["Entity_Timestamp"], "%Y-%m-%dT%H:%M:%S")
+                if (last_dtime <= self.boottime):
+                    list_prop = self.get_properties(reg_entity_id)
+                    tmp_dict = self.merge_prop_dict_list(tmp_dict, list_prop)
+            if ((('entity type' in tmp_dict) and (tmp_dict["entity type"] == entity_type)) and
```

Comment:
Can you please explain this condition in comments in the code? It is very hard to understand.

Reply:
done